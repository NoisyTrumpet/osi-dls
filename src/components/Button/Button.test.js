import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import {
  toBeInTheDocument,
  toHaveClass,
  toHaveAttribute,
} from '@testing-library/jest-dom/matchers';
import { Button } from './';

expect.extend({
  toBeInTheDocument,
  toHaveClass,
  toHaveAttribute,
});

describe('Button', () => {
  test('renders Button component', () => {
    const { container } = render(<Button>Label</Button>);
    expect(screen.getByRole('button')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should have a class of "btn"', () => {
    render(<Button>Label</Button>);
    expect(screen.getByText('Label').closest('button')).toHaveClass('btn');
  });

  test('renders a label within the button', () => {
    const { container, getByText } = render(<Button>Label</Button>);
    expect(getByText('Label')).toBeTruthy();
    expect(container.firstChild).toMatchInlineSnapshot(`
      <button
        class="btn btn-primary"
        type="button"
      >
        
        <span
          class="typography typography-cta color-brand-white"
        >
          Label
        </span>
        
      </button>
    `);
  });

  test('calls "onClick" prop on button click', () => {
    const onClick = jest.fn();
    const { getByRole } = render(<Button click={onClick}>Label</Button>);

    fireEvent.click(getByRole('button'));
    expect(onClick).toHaveBeenCalled();
  });
});
