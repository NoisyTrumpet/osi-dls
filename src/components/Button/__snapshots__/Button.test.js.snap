// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders Button component 1`] = `
<button
  class="btn btn-primary"
  type="button"
>
  
  <span
    class="typography typography-cta color-brand-white"
  >
    Label
  </span>
  
</button>
`;
